components/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -521,6 +537,123 @@ private void handleLogoutRequest(HttpServletRequest request, HttpServletResponse
         }
     }
 
+    /**
+     * Validate the wreply url sent in the logout request with the configured logout wreply url.
+     * The user is redirected to an error page upon validation failure.
+     *
+     * @param request   Logout request.
+     * @throws PassiveSTSException Error in logout url validation.
+     */
+    private void validateLogoutURL(HttpServletRequest request)
+            throws PassiveSTSException {
+
+        String wreplyFromReq = getAttribute(request.getParameterMap(), PassiveRequestorConstants.REPLY_TO);
+        if (StringUtils.isNotBlank(wreplyFromReq)) {
+            String configuredWreply = getConfiguredWreplyUrl(request);
+            if (!wreplyFromReq.equals(configuredWreply)) {
```

Comment:
`wreplyFromReq` can this be a null value?

Reply:
There is a null check in place in line 551 to check that.